extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -236,12 +241,12 @@ AdditionalIndexedClassesBuildItem addPersistenceUnitAnnotationToIndex() {
     @BuildStep
     public void enrollBeanValidationTypeSafeActivatorForReflection(Capabilities capabilities,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
-        if (capabilities.isPresent(Capability.HIBERNATE_VALIDATOR)) {
-            reflectiveClasses.produce(ReflectiveClassBuildItem.builder("org.hibernate.boot.beanvalidation.TypeSafeActivator")
-                    .methods().fields().build());
-            reflectiveClasses.produce(ReflectiveClassBuildItem.builder(BeanValidationIntegrator.JAKARTA_BV_CHECK_CLASS)
-                    .constructors(false).build());
-        }
+        // Accessed in org.hibernate.boot.beanvalidation.BeanValidationIntegrator.loadTypeSafeActivatorClass
+        reflectiveClasses.produce(ReflectiveClassBuildItem.builder("org.hibernate.boot.beanvalidation.TypeSafeActivator")
+                .methods().fields().build());
+        // Accessed in org.hibernate.boot.beanvalidation.BeanValidationIntegrator.isBeanValidationApiAvailable
+        reflectiveClasses.produce(ReflectiveClassBuildItem.builder(BeanValidationIntegrator.JAKARTA_BV_CHECK_CLASS)
+                .constructors(false).build());
```

Comment:
Alternatively, we could skip this code when the `HIBERNATE_VALIDATOR` capability is not present:

https://github.com/quarkusio/quarkus/blob/b4b93ada873b74250112ec0f70362b17af59c3df/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/boot/FastBootMetadataBuilder.java#L479

And then we'd be able to keep the `if` in this method, because `BeanValidationIntegrator.isBeanValidationApiAvailable` wouldn't be called when the `if` condition is `false`.

Though maybe that "optimization" is not worth it, I don't know.

Reply:
Seems reasonable, but I don't know how to check if the validator capability is present.